[SPARK-10775][UI]add search keywords in history page ui

lianhuiwang opened the pull request:
add search button in history page ui that we can search some applications of keywords,example: date or time.The ui is as the following photo:
![historyserver-ui-search](https://cloud.githubusercontent.com/assets/545478/10047463/e26f10e6-6240-11e5-82f5-b54cf580d954.png)

lianhuiwang committed:
add search keywords in history page ui

AmplabJenkins commented:
Merged build triggered.

AmplabJenkins commented:
Merged build started.

SparkQA commented:
[Test build #42908 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/42908/consoleFull) for   PR 8885 at commit [`5f5be5b`](https://github.com/apache/spark/commit/5f5be5bcce29acfff2fe84f2e363afbd8d7ffde6).

SparkQA commented:
[Test build #42908 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/42908/console) for   PR 8885 at commit [`5f5be5b`](https://github.com/apache/spark/commit/5f5be5bcce29acfff2fe84f2e363afbd8d7ffde6).
 * This patch **passes all tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/42908/
Test PASSed.

andrewor14 commented:
@lianhuiwang would you mind rebasing to master?

andrewor14 commented:
@sarutak @vanzin

sarutak commented:
I'll inspect the details but I'm not sure whether it's useful to enable to search by app id.

sarutak commented on core/src/main/scala/org/apache/spark/deploy/history/HistoryPage.scala:
```diff
@@ -74,19 +80,27 @@ private[history] class HistoryPage(parent: HistoryServer) extends WebUIPage("")
 
               <h4>
+                <form action="" method="GET">
+                  Search: <input type="text" name="query" value={requestedQuery} />
```
It's difficult for users to know the targets searched by keywords. How about set initial message "Input App Names" or like that to the text field and remove "Search" label.
It's just one idea. Someone might have better idea.

sarutak commented on core/src/main/scala/org/apache/spark/deploy/history/HistoryPage.scala:
```diff
@@ -74,19 +80,27 @@ private[history] class HistoryPage(parent: HistoryServer) extends WebUIPage("")
+                <form action="" method="GET">
+                  Search: <input type="text" name="query" value={requestedQuery} />
+                  <input type="hidden" name="showIncomplete" value={requestedIncomplete.toString} />
+                  <input type="submit" value="Submit" />
```
I think "search" is better than "Submit".

lianhuiwang commented:
@andrewor14 Yes, I will rebase to master.

lianhuiwang committed:
Merge branch 'apache-master' into SPARK-10775

Conflicts:
	core/src/main/scala/org/apache/spark/deploy/history/HistoryPage.scala

lianhuiwang committed:
rename Submit to Search

lianhuiwang committed:
Merge branch 'apache/master' into spark-10775

lianhuiwang committed:
Revert "rename Submit to Search"

This reverts commit bff49b8f3b6a3e3562253fabfcbf2a11af4d9c88.

lianhuiwang committed:
fix minor style

lianhuiwang committed:
update minor code

lianhuiwang commented:
@andrewor14 Now i think that's OK. Can you look at it? Thanks.

SparkQA commented:
**[Test build #47825 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/47825/consoleFull)** for PR 8885 at commit [`16965b6`](https://github.com/apache/spark/commit/16965b681e85276052abba8f6ffd1ff7dc17e3ea).

AmplabJenkins commented:
Merged build finished. Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/47822/
Test FAILed.

lianhuiwang commented:
test it please

AmplabJenkins commented:
Merged build finished. Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/47823/
Test FAILed.

SparkQA commented:
**[Test build #47828 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/47828/consoleFull)** for PR 8885 at commit [`16965b6`](https://github.com/apache/spark/commit/16965b681e85276052abba8f6ffd1ff7dc17e3ea).

SparkQA commented:
**[Test build #47825 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/47825/consoleFull)** for PR 8885 at commit [`16965b6`](https://github.com/apache/spark/commit/16965b681e85276052abba8f6ffd1ff7dc17e3ea).
 * This patch passes all tests.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/47825/
Test PASSed.

SparkQA commented:
**[Test build #47828 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/47828/consoleFull)** for PR 8885 at commit [`16965b6`](https://github.com/apache/spark/commit/16965b681e85276052abba8f6ffd1ff7dc17e3ea).
 * This patch **fails PySpark unit tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/47828/
Test FAILed.

lianhuiwang committed:
Merge branch 'apache-master' into spark-10775-2

lianhuiwang committed:
address sarutak's comments

lianhuiwang committed:
fix minor name

SparkQA commented:
**[Test build #47880 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/47880/consoleFull)** for PR 8885 at commit [`a3f6885`](https://github.com/apache/spark/commit/a3f6885a90d970c8d128522a40df496dfc1612a7).

AmplabJenkins commented:
Merged build finished. Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/47879/
Test FAILed.

lianhuiwang commented:
@sarutak Thanks. I have addressed your comments.Now UI is as following:
![history-ui](https://cloud.githubusercontent.com/assets/545478/11859513/a371acf2-a4a6-11e5-9f67-b16d9c372956.png)

SparkQA commented:
**[Test build #47880 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/47880/consoleFull)** for PR 8885 at commit [`a3f6885`](https://github.com/apache/spark/commit/a3f6885a90d970c8d128522a40df496dfc1612a7).
 * This patch **fails PySpark unit tests**.
 * This patch merges cleanly.
 * This patch adds the following public classes _(experimental)_:\n  * `public class JavaBisectingKMeansExample `\n  * `class CrossValidator(Estimator, HasSeed):`\n

AmplabJenkins commented:
Merged build finished. Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/47880/
Test FAILed.

lianhuiwang commented:
test this please.

SparkQA commented:
**[Test build #47897 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/47897/consoleFull)** for PR 8885 at commit [`a3f6885`](https://github.com/apache/spark/commit/a3f6885a90d970c8d128522a40df496dfc1612a7).

SparkQA commented:
**[Test build #47897 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/47897/consoleFull)** for PR 8885 at commit [`a3f6885`](https://github.com/apache/spark/commit/a3f6885a90d970c8d128522a40df496dfc1612a7).
 * This patch passes all tests.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/47897/
Test PASSed.

lianhuiwang commented:
@andrewor14 @sarutak Now all tests have passed.

tgravescs commented:
so in https://issues.apache.org/jira/browse/SPARK-10873 we are looking at changing to use the jquery datatables.  This would give us a bunch of things including a search box for free.  We are close to having a prototype of the history server working.

sarutak commented:
@lianhuiwang #10648 has already merged and I think it includes features proposed in this PR. So, do you mind closing this PR?

lianhuiwang commented:
@sarutak Yes, I will close this PR. Thanks.
